Examples/FirebaseDemo/RemoteConfigKey.swift

```diff
@@ -10,39 +11,108 @@ extension SharedReaderKey {
 
 struct RemoteConfigKey<Value: Decodable & Sendable>: SharedReaderKey {
   let key: String
+  let remoteConfig: any RemoteConfigClient
+
+  init(key: String) {
+    self.key = key
+    @Dependency(\.remoteConfig) var remoteConfig
+    self.remoteConfig = remoteConfig
+  }
+
   var id: some Hashable { key }
-  @Dependency(RemoteConfigDependencyKey.self) var remoteConfig
+
   func load(context _: LoadContext<Value>, continuation: LoadContinuation<Value>) {
-    remoteConfig.fetchAndActivate { changed, error in
-      if let error {
-        continuation.resume(throwing: error)
-      } else {
-        continuation.resume(with: Result { try remoteConfig.configValue(forKey: key).decoded() })
-      }
-    }
+    remoteConfig.fetch(key: key, completion: continuation.resume(with:))
   }
   func subscribe(
     context _: LoadContext<Value>, subscriber: SharedSubscriber<Value>
   ) -> SharedSubscription {
-    let registration = remoteConfig.addOnConfigUpdateListener { update, error in
-      guard error == nil else { return }
-      remoteConfig.activate { changed, error in
-        guard error == nil else { return }
-        subscriber.yield(with: Result { try remoteConfig.configValue(forKey: key).decoded() })
-      }
-    }
+    nonisolated(unsafe) let cancellable = remoteConfig.addUpdateListener(
```

Comment:
Does `@preconcurrency import Combine` silence this issue?

Reply:
it does!